extensions/ext/GL_EXT_vulkan_glsl_relaxed.txt

```diff
@@ -0,0 +1,177 @@
+    It is also not enabled through use of a profile or #version.  The intended
+    level of GLSL/ESSL features, independent from Vulkan-specific usage, comes
+    from the traditional use of #version, profile, and #extension.
+
+    Instead, use of this extension is an effect of using a GLSL front-end in a
+    mode that has it generate SPIR-V for Vulkan.  Such tool use is outside the
+    scope of using the Vulkan API and outside the definition of GLSL and this
+    extension. See the documentation of the compiler to see how to request
+    generation of SPIR-V for Vulkan using these rules.
+
+    When a front-end is used to accept this extension, it must error check and
+    reject shaders not adhering to this specification, and accept those that
+    do. Implementation-dependent maximums and capabilities are supplied to, or
+    part of, the front-end, so it can do error checking against them.
+             
+    Default Uniforms
+    ---------------
+    
+    All default uniforms will be silently grouped into a single anonymous
+    uniform block. The block name of the block, the backing, set, and binding
+    number can all be controlled via options to the GLSL front-end. If no
+    name is given, the block name will be 'gl_DefaultUniformBlock'. 
+
+    Shared and Packed Block Layouts
+    -------------------------------
+    
+    If a block is declared with shared or packed layouts, those layouts will
+    be ignored and instead the default layout for that block type will be used.
+    
+    gl_VertexID and gl_InstanceID
+    -----------------------------
+
+    Built-in variables gl_VertexID and gl_InstanceID are accepted, but are
+    simply aliases to gl_VertexIndex and gl_InstanceIndex which were introduced
+    in GL_EXT_vulkan_glsl. They will include the base offset like the
+    GL_EXT_vulkan_glsl behavior.
+    
+    Atomic Integers
+    ---------------
+    
+    atomic_uint declarations and functions inteacting with them will be
+    silently converted to use storage buffers and related atomic functions.
+    For each binding used for the atomic_uint variables, a different 
+    buffer block will be allocated and the variables will be placed
+    within that buffer using the variable's offset. The block name of
+    the block and set number can all be controlled via options to the
+    GLSL front-end. If no name is given, the block name will start with
+    'gl_AtomicBufferBlock'. The block name will suffixed with '_<bindingNumber>'.
+    Where <bindingNumber> is the original binding specified for the
+    atomic_uints contained within that block. The actual binding of the
+    block may be different than what was originally specified in the shader.
+    Reflection should be used to find the final binding locations of these
+    blocks.
+   
+Changes to Chapter 3 of the OpenGL Shading Language Specification
+
+    Move the following keywords in section 3.6 Keywords back out of
```

Comment:
Here is where it is important to know what specification these modification instructions are against.